src/contrib/persistence/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -0,0 +1,421 @@
+                        r.PersistenceId == persistenceId);
+                queryable = queryable.Union(nextQuery.Select(md =>
+                        new
+                        {
+                            SequenceNumber = md.SequenceNumber,
+                            PersistenceId = md.PersistenceId
+                        }));
+            }
+
+            return queryable.OrderByDescending(r => r.SequenceNumber)
+                .Select(r => r.SequenceNumber).Take(1);
+        }
+
+        public async Task<Done> Update(string persistenceId, long sequenceNr,
+            object payload)
+        {
+            var write = new Persistent(payload, sequenceNr, persistenceId);
+            var serialize = Serializer.Serialize(write);
+            if (serialize.IsSuccess)
+            {
+                throw new ArgumentException(
+                    $"Failed to serialize {write.GetType()} for update of {persistenceId}] @ {sequenceNr}",
+                    serialize.Failure.Value);
+            }
+
+            using (var db = _connectionFactory.GetConnection())
+            {
+                await db.GetTable<JournalRow>()
+                    .Where(r =>
+                        r.persistenceId == persistenceId &&
+                        r.sequenceNumber == write.SequenceNr)
+                    .Set(r => r.message, serialize.Get().message)
+                    .UpdateAsync();
+                return Done.Instance;
+            }
+        }
+
+        public async Task<long> HighestSequenceNr(string persistenceId,
+            long fromSequenceNr)
+        {
+            using (var db = _connectionFactory.GetConnection())
+            {
+                return await MaxSeqNumberForPersistenceIdQuery(db,
+                    persistenceId,
+                    fromSequenceNr).FirstOrDefaultAsync();
+            }
+        }
+
+        
+
+        public override
+            async Task<Source<Util.Try<ReplayCompletion>, NotUsed>>
+            Messages(DataConnection db, string persistenceId,
+                long fromSequenceNr, long toSequenceNr,
+                long max)
+        {
+
+            {
+                IQueryable<JournalRow> query = db.GetTable<JournalRow>()
+                    .Where(r =>
+                        r.persistenceId == persistenceId &&
+                        r.sequenceNumber >= fromSequenceNr &&
+                        r.sequenceNumber <= toSequenceNr &&
+                        r.deleted == false)
+                    .OrderBy(r => r.sequenceNumber);
+                if (max <= int.MaxValue)
+                {
+                    query = query.Take((int) max);
+                }
+
+                
+                //TODO: Is there a better way to do this async?
+                //return Source
+                //    .FromObservable(query.AsAsyncEnumerable().ToObservable())
+                
+                return Source.From(await query.ToListAsync())
```

Comment:
FWIW, there is potential to use -real- IAsyncEnumerable here in future, Linq2Db supports it, I just couldn't find an abstraction in Akka.Streams to utilize with actual benefit